app/jobs/scheduled/clean_up_unused_staged_users.rb

```diff
@@ -6,11 +6,21 @@ class CleanUpUnusedStagedUsers < Jobs::Scheduled
     def execute(args)
       destroyer = UserDestroyer.new(Discourse.system_user)
 
-      User.joins(:user_stat)
-          .where(staged: true)
-          .where("users.created_at < ?", 1.year.ago)
-          .where("user_stats.post_count = 0")
-          .find_each { |user| destroyer.destroy(user) }
+      User.joins("LEFT JOIN posts ON posts.user_id = users.id")
+        .where("posts.user_id IS NULL")
+        .where(staged: true)
+        .where("users.created_at < ?", 1.year.ago)
+        .find_each do |user|
+
+        begin
+          destroyer.destroy(user)
+        rescue => e
```

Comment:
`PostDestroyer` would raise an exception if certain conditions are not met. Therefore, we don't want to have the entire scheduled job blow up just because we are not able to destroy a user.